### PR TITLE
fix: broken logic, refactoring, and dep updates

### PR DESCRIPTION
**Pull request by dtbuchholz**
## Summary

This PR implements a few things:
- Updates deps to latest SDK and moves dependencies/devDependencies accordingly.
- Refactors to conform with latest SDK return type for reads with `createProcessor`.
- Provides a more secure way (via chatgpt) to use the encryption plugin (even though it's "not recommended," it might as well provide a more robust impl.)
- Fixes behavior with `Pinner` class & `ipfs-http-client` where pinning wasn't working
  - For local-only pinning, the logic was still checking for remote pinning services and always errored out.
  - `pinToProvider` and `pinToLocal` now must be instantiated an IPFS node gateway URL before templating is possible.
- For the `Pinner`, also allow for raw strings _or_ Uint8Arrays to be pinned, retrieving the CID contents uses `cat`/streams to get the original text contents.
- Refactored the `truncate` method to actually be useful where it truncates text at 1024 bytes to align with table cell limits—i.e., an alternative to using the IPFS plugin & CIDs.

## Details

Since the API changes for `pinToProvider` and `pinToLocal` as well as behavior for others, this needs a major bump.

### CODE REVIEW

**Comment by dtbuchholz** [comment]
@joewagner could you maybe give this a passover as well? it basically cleans up existing logic to make it more usable. i've incorporated these changes in the docs site PR, too.

File: package.json [hunk 1]
```diff
@@ -58,32 +58,33 @@
...
-    "add-js-extension": "^1.0.4",
+    "buffer": "^6.0.3",
     "crypto-js": "3.1.9-1",
-    "esbuild": "^0.17.18",
-    "eslint": "^8.40.0",
-    "eslint-plugin-import": "^2.27.5",
-    "ipfs-http-client": "^60.0.1",
-    "prettier": "^2.8.8",
-    "ts-node": "^10.9.1"
+    "ipfs-http-client": "^60.0.1"
```
Comment (joewagner) [inline package.json:87]: @dtbuchholz Do you want to add yourself as a contributor?

**Review by joewagner — DISMISSED** [review]
@dtbuchholz Thanks for getting this working.  I added two inline comments, one is a spelling error, otherwise looks great!

**Comment by dtbuchholz** [comment]
@joewagner good catches! updated, just need a new 👍 and then i'll publish

**Review by joewagner — APPROVED** [review]

**Pull request merged**